### PR TITLE
[Draft] feat: partial switch to use client script proxy

### PR DESCRIPTION
**Pull request by maijalintunokka**
Early version of attempting to use the nosto-js-sdk in the nosto-react project.

To run locally:
- `yarn link` in the `nosto-js-sdk` folder.
- `yarn link nosto-js` in the `nosto-react` folder.

### CODE REVIEW

File: spec/utils.tsx [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-  if (!window.nostojs) {
-    window.nostojs = (cb: (api: NostoClient) => void) => {
-      (window.nostojs.q = window.nostojs.q || []).push(cb)
-    }
-  }
-  window.nostojs(api => api.listen(event, req => requests.push(req)))
+  initNostoStub()
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-expect-error
+  nostojs(api => api.listen(event as Parameters<API['listen']>[0], (req) => {
```
Comment (timowestnosto) [inline spec/utils.tsx:15]: move event type change to top level function?

Comment (maijalintunokka) [reply spec/utils.tsx:15]: This is not a real PR so far. More of a "here's how you could theoretically do it".

File: spec/useLoadClientScript.spec.tsx [hunk 2]
```diff
@@ -12,7 +13,7 @@ function loadClientScript(merchant: string) {
   script.async = true
   const promise = new Promise<void>(resolve => {
     script.onload = () => {
-      window.nosto?.reload({ site: "localhost" })
+      getNostoWindowOrDefault()?.reload({ site: "localhost" })
```
Comment (timowestnosto) [inline spec/useLoadClientScript.spec.tsx:16]: better expose reload directly ?

File: spec/useLoadClientScript.spec.tsx [hunk 3]
```diff
@@ -33,8 +34,8 @@ describe("useLoadClientScript", () => {
 
     hook.rerender()
     expect(hook.result.current.clientScriptLoaded).toBe(true)
-    expect(window.nosto).toBeDefined()
-    expect(getScriptSources()).toEqual([`http://connect.nosto.com/include/${testAccount}`])
+    expect(getNostoWindow()).toBeDefined()
```
Comment (timowestnosto) [inline spec/useLoadClientScript.spec.tsx:37]: should we expose isLoaded() instead ?

File: src/types.ts [hunk 4]
```diff
@@ -13,15 +15,7 @@ declare global {
...
-export interface NostoClient {
-  setAutoLoad(autoload: boolean): void
-  defaultSession(): Session
-  listen(event: string, callback: (data: unknown) => void): void
-  placements: {
-    getPlacements(): string[]
-    injectCampaigns(recommendations: Record<string, unknown>): void
-  }
-}
+export type NostoClient = API
```
[the remark this answers is not in the document]
Comment (maijalintunokka) [reply src/types.ts]: Removed now

File: spec/utils.tsx (hunk 1 above)
Comment (maijalintunokka) [reply spec/utils.tsx]: I've figured out why there were some issues here. It's now clean 🙂

File: spec/useLoadClientScript.spec.tsx (hunk 3 above)
Comment (maijalintunokka) [reply spec/useLoadClientScript.spec.tsx]: It was already exposed, I guess I just forgot to use it here. Replaced. 🙂

File: spec/useLoadClientScript.spec.tsx (hunk 2 above)
Comment (maijalintunokka) [reply spec/useLoadClientScript.spec.tsx]: Done 👍

**Review by timowestnosto — APPROVED** [review]

**Pull request merged**